app/models/host.rb

```diff
@@ -1568,6 +1524,30 @@ def scan_from_queue(taskid=nil)
     _log.info("Scanning #{log_target}...Complete - Timings: #{t.inspect}")
   end
 
+  def scan_refreshes(log_target, task)
+    # Firewall Rules and Advanced Settings go through EMS so we don't need Host credentials
+    scan_refresh("Firewall Rules", log_target, task) { refresh_firewall_rules }
+    scan_refresh("Advanced Settings", log_target, task) { refresh_advanced_settings }
+    scan_refresh("IPMI information", log_target, task) { refresh_ipmi } unless ext_management_system
+  end
+
+  def scan_refreshes_with_connection(log_target, task, ssu)
+    scan_refresh("Patches", log_target, task, ssu) { |ssu| refresh_patches(ssu) }
+    scan_refresh("Services", log_target, task, ssu) { |ssu| refresh_services(ssu) }
+    scan_refresh("Linux Packages", log_target, task, ssu) { |ssu| refresh_services(ssu) }
+    scan_refresh("User Groups", log_target, task, ssu) { |ssu| refresh_user_groups(ssu) }
+    scan_refresh("SSH Config", log_target, task, ssu) { |ssu| refresh_ssh_config(ssu) }
+    scan_refresh("FS Files", log_target, task, ssu) { |ssu| refresh_fs_files(ssu) }
+  end
+
+  def scan_refresh(title, log_target, task, ssu = nil)
+    _log.info("Refreshing #{title} for #{log_target}")
+    task.update_status("Active", "Ok", "Refreshing #{title}") if task
+
+    benchmark_name = "refresh_#{title.parameterize.underscore}".to_sym
+    Benchmark.realtime_block(benchmark_name) { ssu ? yield(ssu) : yield }
```

Comment:
Well, it yields two forms ... one with the `ssu` param (hate that name, but kept it for consistency with the things it's yielding to) and one without the `ssu` param.

I really wanted a better way to do this, but couldn't see one that didn't result in ridiculous repeated code (that I was trying to get away from in the first place).

however, if you mean I don't have to wrap the `ssu` in parens, as in:

```ruby
  { ssu ? yield ssu : yield }
```

I actually tried that first and got a syntax error around `ssu` since it couldn't parse what was going on there without the parens.